### PR TITLE
Deprecate nearby_ids_exact in favour of nearby_ids with a search keyword

### PR DESCRIPTION
**Pull request by Tortar**
This makes the API cleaner

### CODE REVIEW

**Comment by Tortar** [comment]
This fixes also #1043

**Comment by Tortar** [comment]
Actually there is still something to take care because sometimes we pass `nearby_ids_exact` to some other functions

**Comment by Tortar** [comment]
If tests pass, it should be okay now

File: src/deprecations.jl [hunk 1]
```diff
@@ -379,4 +379,15 @@ until(s, f, model) = !f(model, s)
...
+
+# Do the standard extensions for `_exact` as in space API
+function nearby_ids_exact(agent::AbstractAgent, model::ABM, r = 1)
+    @warn "`nearby_ids_exact` is deprecated in favor of `nearby_ids(...; search=:exact)`." maxlog=1
+    all = nearby_ids_exact(agent.pos, model, r)
+    Iterators.filter(i -> i ≠ agent.id, all)
+end
+function nearby_agents_exact(a, model, r=1)
+    @warn "`nearby_agents_exact` is deprecated in favor of `nearby_agents(...; search=:exact)`." maxlog=1
+    return (model[id] for id in nearby_ids_exact(a, model, r))
```
Comment (Datseris) [inline src/deprecations.jl:392]: Shouldn't this source code just call the `nearby_ids(...; search = :exact)` instead of what it does now?

**Review by Datseris — APPROVED** [review]
Seems good, but bump minor version (and add changelog) and release new version after merge!

File: src/deprecations.jl (hunk 1 above)
Comment (Tortar) [reply src/deprecations.jl]: yes, did it

**Comment by Datseris** [comment]
I wonder if we can increase the performance of the `nearest_neighbor` function... We go thourhg all agents in the list.

On the other hand, we do near searches fundamentally differently from e.g., a tree structure, so it's not like we would ever have nearby agents sorted by distance. Anyways, merging and tagging.

**Comment by Tortar** [comment]
I think it can be quite improved by the user, because decreasing the radius implies searching less neighbors, but it is model dependent, one can try little radius before big ones

**Pull request merged**